src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -162,7 +168,21 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
       primaryLabel: {
         type: LabelSignalEnumType,
         description: "Primary label signal available to collector",
-        resolve: (artwork, {}, ctx) => getPrimaryLabel(artwork, ctx),
+        args: {
+          ignore: {
+            type: GraphQLList(LabelSignalEnumType),
+            description: "Signals to ignore",
+          },
+        },
+        resolve: (artwork, args, ctx) => {
+          if (args.ignore?.length > 0) {
+            const uniqueIgnoredLabels = new Set(args.ignore)
+            if (uniqueIgnoredLabels.size !== args.ignore.length) {
+              throw new Error("Duplicate values found in ignore argument")
+            }
+          }
```

Comment:
I imagined a long list of ignored labels causing problems for us. Not sure if this is actually necessary. Started with a uniqueness test, now i'm just limiting the total length to the length of possible enum values.